### PR TITLE
Apache Camel Server Generator

### PR DESCRIPTION
**Pull request by carnevalegiacomo**
Generate models and Apache Camel Rest Definitions (Java DSL).

Features (Api first):
 * Bean Validation
 * Perform Bean Validation
 * Binding mode

Generation:
  * Maven Spring Boot Project
  * Mock
  * Unit Test

### PR checklist
 
- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] Pull Request title clearly describes the work in the pull request and Pull Request description provides details about how to validate the work. Missing information here may result in delayed response from the community.
- [x] Run the following to [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) and update samples:
  ```
  ./mvnw clean package 
  ./bin/generate-samples.sh
  ./bin/utils/export_docs_generators.sh
  ``` 
  Commit all changed files. 
  This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit as it would merge with master. 
  These must match the expectations made by your contribution. 
  You may regenerate an individual generator by passing the relevant config(s) as an argument to the script, for example `./bin/generate-samples.sh bin/configs/java*`. 
  For Windows users, please run the script in [Git BASH](https://gitforwindows.org/).
- [x] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master` (5.3.0), `6.0.x`
- [x] If your PR is targeting a particular programming language, @mention the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) members, so they are more likely to review the pull request.

### CODE REVIEW

**Comment by wing328** [comment]
Thanks for the PR but your commit (as shown in the Commits tab) is not linked to your Github account, which means this PR won't count as your contribution in https://github.com/OpenAPITools/openapi-generator/graphs/contributors.

Let me know if you need help fixing it.

Ref: https://github.com/OpenAPITools/openapi-generator/wiki/FAQ#how-can-i-update-commits-that-are-not-linked-to-my-github-account

**Comment by wing328** [comment]
@carnevalegiacomo I'll later rename the generator to `java-camel` (in a a separate PR) to follow the convention `{lang}-{framework-name}` that we've been using. Let me know if you've any concern with that.

**Comment by carnevalegiacomo** [comment]
Hi
I renamed the generator to java-camel

Giacomo

**Comment by wing328** [comment]
Tested locally and the result is good:
```
[INFO] --- spring-boot-maven-plugin:2.6.2:repackage (default) @ openapi-camel ---
Downloading from central: https://repo.maven.apache.org/maven2/org/springframework/boot/spring-boot-buildpack-platform/2.6.2/spring-boot-buildpack-platform-2.6.2.pom
Downloaded from central: https://repo.maven.apache.org/maven2/org/springframework/boot/spring-boot-buildpack-platform/2.6.2/spring-boot-buildpack-platform-2.6.2.pom (3.4 kB at 10 kB/s)
Downloading from central: https://repo.maven.apache.org/maven2/org/springframework/boot/spring-boot-loader-tools/2.6.2/spring-boot-loader-tools-2.6.2.pom
Downloaded from central: https://repo.maven.apache.org/maven2/org/springframework/boot/spring-boot-loader-tools/2.6.2/spring-boot-loader-tools-2.6.2.pom (2.3 kB at 7.0 kB/s)
Downloading from central: https://repo.maven.apache.org/maven2/org/springframework/boot/spring-boot-buildpack-platform/2.6.2/spring-boot-buildpack-platform-2.6.2.jar
Downloading from central: https://repo.maven.apache.org/maven2/org/springframework/boot/spring-boot-loader-tools/2.6.2/spring-boot-loader-tools-2.6.2.jar
Downloaded from central: https://repo.maven.apache.org/maven2/org/springframework/boot/spring-boot-buildpack-platform/2.6.2/spring-boot-buildpack-platform-2.6.2.jar (242 kB at 642 kB/s)
Downloaded from central: https://repo.maven.apache.org/maven2/org/springframework/boot/spring-boot-loader-tools/2.6.2/spring-boot-loader-tools-2.6.2.jar (249 kB at 627 kB/s)
[INFO] Replacing main artifact with repackaged archive
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  01:35 min
[INFO] Finished at: 2022-01-04T15:37:13+08:00
[INFO] ------------------------------------------------------------------------
```

File: modules/openapi-generator/src/main/resources/JavaSpring/pojo.mustache [hunk 1]
```diff
@@ -25,6 +25,7 @@ public class {{classname}} {{#parent}}extends {{{.}}}{{/parent}}{{^parent}}{{#ha
   {{/jackson}}
   {{#gson}}
   @SerializedName("{{baseName}}")
+  @SerializedName("{{baseName}}")
```
Comment (wing328) [inline modules/openapi-generator/src/main/resources/JavaSpring/pojo.mustache:28]: @carnevalegiacomo is this line added by mistake?

Comment (carnevalegiacomo) [reply modules/openapi-generator/src/main/resources/JavaSpring/pojo.mustache:28]: yes

Comment (wing328) [reply modules/openapi-generator/src/main/resources/JavaSpring/pojo.mustache:28]: can you please remove it when you've time?

Comment (carnevalegiacomo) [reply modules/openapi-generator/src/main/resources/JavaSpring/pojo.mustache]: https://github.com/OpenAPITools/openapi-generator/pull/11162/commits/5813fb18440c2c062752075b39bb3f710d8c07d0

**Comment by wing328** [comment]
LGTM. Thanks for the new generator. 

I'll file another PR later with some minor enhancements.

**Pull request merged**